policy/modules/contrib/qatlib.te

```diff
@@ -0,0 +1,52 @@
+policy_module(qatlib, 1.0.0)
+
+########################################
+#
+# Declarations
+#
+
+type qatlib_t;
+type qatlib_exec_t;
+init_daemon_domain(qatlib_t, qatlib_exec_t)
+
+type qatlib_conf_t;
+files_config_file(qatlib_conf_t)
+
+type qatlib_unit_file_t;
+systemd_unit_file(qatlib_unit_file_t)
+
+type qatlib_var_run_t;
+files_pid_file(qatlib_var_run_t)
+
+########################################
+#
+# qatlib local policy
+#
+allow qatlib_t self:fifo_file rw_fifo_file_perms;
+allow qatlib_t self:unix_stream_socket create_stream_socket_perms;
+
+allow qatlib_t qatlib_unit_file_t:file read_file_perms;
+
+manage_dirs_pattern(qatlib_t, qatlib_var_run_t, qatlib_var_run_t)
+manage_files_pattern(qatlib_t, qatlib_var_run_t, qatlib_var_run_t)
+files_pid_filetrans(qatlib_t, qatlib_var_run_t, { dir file } )
+
+corecmd_exec_shell(qatlib_t)
+corecmd_exec_bin(qatlib_t)
+
+dev_read_sysfs(qatlib_t)
+
+domain_use_interactive_fds(qatlib_t)
+
+optional_policy(`
+        auth_read_passwd_file(qatlib_t)
+')
+
+optional_policy(`
+	miscfiles_read_localization(qatlib_t)
+')
+
+optional_policy(`
+	systemd_search_unit_dirs(qatlib_t)
```

Comment:
Why does it need to search units directory? Is it because it wants to access some particular unit?

Reply:
It is using awk command for checking systemd configuration file.

AVC:
```
type=PATH msg=audit(1674238318.319:806): item=0 name="/usr/lib/systemd/system/qat.service" inode=159310 dev=fc:02 mode=0100644 ouid=0 ogid=0 rdev=00:00 obj=system_u:object_r:qatlib_unit_file_t:s0 nametype=NORMAL cap_fp=0 cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0
type=CWD msg=audit(1674238318.319:806): cwd="/"
type=SYSCALL msg=audit(1674238318.319:806): arch=c000003e syscall=257 success=yes exit=3 a0=ffffff9c a1=55dcf764e110 a2=0 a3=0 items=1 ppid=2592 pid=2593 auid=4294967295 uid=0 gid=992 euid=0 suid=0 fsuid=0 egid=992 sgid=992 fsgid=992 tty=(none) ses=4294967295 comm="awk" exe="/usr/bin/gawk" subj=system_u:system_r:qatlib_t:s0 key=(null)
type=AVC msg=audit(1674238318.319:806): avc:  denied  { search } for  pid=2593 comm="awk" name="system" dev="vda2" ino=137418 scontext=system_u:system_r:qatlib_t:s0 tcontext=system_u:object_r:systemd_unit_file_t:s0 tclass=dir permissive=1
```